src/libraries/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Windows.cs

```diff
@@ -102,13 +102,21 @@ internal bool SkipCompletionPortOnSuccess
         }
 
         /// <returns>Returns whether operations were canceled.</returns>
-        private bool OnHandleClose()
+        private unsafe bool OnHandleClose()
         {
             // Keep m_IocpBoundHandle around after disposing it to allow freeing NativeOverlapped.
             // ThreadPoolBoundHandle allows FreeNativeOverlapped even after it has been disposed.
             if (_iocpBoundHandle != null)
             {
                 Debug.Assert(OperatingSystem.IsWindows());
+
+                // If the handle is owned, on-going async operations will be aborted when the handle is closed.
+                // If we don't own the handle, cancel them explicitly.
+                if (!OwnsHandle)
+                {
+                    Interop.Kernel32.CancelIoEx(handle, null);
```

Comment:
We're cancelling all operations that are performed against the handle here.
If there is another `SafeSocketHandle` for the same raw handle in this process, it may get its operations aborted by this.

For Unix we track the on-going operations performed against the `SafeHandle` in `SocketAsyncContext` so we're only canceling the operations that were performed against that handle.
On Unix we actually don't support different `SafeSocketHandle`s to perform async operations on the same raw handle. `SocketAsyncEngine` holds a map from the raw handle to one `SocketAsyncContext`.

Reply:
I can't.